pip/utils/__init__.py

```diff
@@ -54,6 +55,37 @@ def get_prog():
     return 'pip'
 
 
+def default_user_site():
+    # Avoid circular import, the running_under_virtualenv should probably be
+    # in pip.utils anyways TBH.
+    from pip.locations import running_under_virtualenv, distutils_scheme
+
+    # If we're running inside of a virtual environment, we do not want to
+    # install to the --user directory.
+    if running_under_virtualenv():
+        return False
+
+    # If the Python we're running under does not have their user packages
+    # enabled then we do not want to install to the user directory since it
+    # may or may not work or exist.
+    if not site.ENABLE_USER_SITE:
+        return False
+
+    # If any of our potentional locations for writing files is not writable by
+    # us then we want to use the --user scheme instead of the --global scheme.
+    # TODO: We should figure out a way to make this work for --root and
```

Comment:
if someone intentionally specifies a `--root`, but they don't have permissions, it should just fail I think, and never switch to a user install.  same thing when specifying custom `--install-option` arguments.

note that our `distutils_scheme` function reads custom locations from distutils config files, so with your implementation as is, you will be switching users to a user install if any of those custom paths are not writable, which I think is misleading.